src/java.base/share/classes/sun/net/www/protocol/https/HttpsClient.java

```diff
@@ -565,6 +563,7 @@ public void afterConnect() throws IOException, UnknownHostException {
                     // will do the spoof checks in SSLSocket.
                     SSLParameters paramaters = s.getSSLParameters();
                     paramaters.setEndpointIdentificationAlgorithm("HTTPS");
+                    paramaters.setServerNames(List.of(new SNIHostName(host)));
```

Comment:
Thank you for taking my comment.  But I may not update line 456-458.  A safe update may be in line 566, for example:
+    // The host has been set for SSLSocketImp previously.
+    if (!(s instanced SSLSocketImpl)) {
          paramaters.setServerNames(List.of(new SNIHostName(host)));
+    }

Reply:
I did the changes as you suggested.